types/src/jwks/mod.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright © Aptos Foundation
+
+use crate::move_any::{Any, AsMoveAny};
+use aptos_crypto_derive::{BCSCryptoHash, CryptoHasher};
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeSet;
+use aptos_crypto::bls12381;
+use move_core_types::account_address::AccountAddress;
+use crate::jwks::rsa::RSA_JWK;
+use crate::jwks::unsupported::UnsupportedJWK;
+
+pub type Issuer = Vec<u8>;
+
+pub fn issuer_from_str(s: &str) -> Issuer {
+    s.as_bytes().to_vec()
+}
+
+/// Move type `0x1::jwks::OIDCProvider` in rust.
+pub struct OIDCProvider {
+    pub name: Issuer,
+    pub config_url: Vec<u8>,
+}
+
+/// Move type `0x1::jwks::SupportedOIDCProviders` in rust.
+pub struct SupportedOIDCProviders {
+    pub providers: Vec<OIDCProvider>,
+}
+
+/// Move type `0x1::jwks::JWK` in rust.
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize, CryptoHasher, BCSCryptoHash)]
+pub struct JWK {
+    variant: Any,
```

Comment:
Is this the jwk pubkey? why call it variant?

Reply:
i should probably merge this PR into its upstream, where move structs are defined....
It's supposed to be an enum which has 2 variant: RSA and unsupported.
But `Any` is closest thing we can have in Move. And we want to keep the same format in rust so it can be loaded as an on-chain config.